partner_lastname_uppercase/models/res_partner.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2023 Coop IT Easy SC
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.model
+    def uppercase_all_lastnames(self):
+        individuals = self.search(
+            [
+                ("is_company", "=", False),
+                ("lastname", "!=", False),
+            ]
+        )
+        for partner in individuals:
+            partner.lastname = partner.lastname.upper()
```

Comment:
note that `str.upper()` [uses the default unicode case folding](https://docs.python.org/3/library/stdtypes.html#str.upper). this works for many languages, but not for all, like turkish where ı/I i/İ pairs work differently (see [cpython issue 17252](https://bugs.python.org/issue17252)). this limitation should be explained in the readme.